### ISSUE TITLE
[Bug]: non-responding worker

### ISSUE THREAD

**Issue by fredrik-w** [aspect-build/rules_js#667]
### What happened?

When migrating away a mono-repo from rules_nodejs to rules_js/rules_ts I seem to hit an issue with workers not responding. It seems to be consistent with doing `bazel test //...` as it fails each time. It builds all modules fine, then when it's time to run the tests it just times out (I assume the timeout is killing the worker). Running in our CI pipe in GCP (shouldn't make any difference).

As you might notice we have some test_utils to make testing easier, and those are only built in the test phase as they are flagged `testonly = True`. I haven't had the time to try and reproduce it locally yet as building the monolith eats all my CPU resources and the computer gets rather unresponsive. And it takes quite some time to build it.

```bash
ERROR: /workspace/modules/mod_a/BUILD.bazel:50:10: Compiling TypeScript project //modules/mod_a:test_util [tsc -p tsconfig.bazel.json] failed: Worker process did not return a WorkResponse:

---8<---8<--- Start of log, file at /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/bazel-workers/worker-3-TsProject.log ---8<---8<---
Running TsProject as a Bazel worker
/builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/monolith/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/npm_typescript/tsc_worker.sh: line 501: 10759 Killed                  "$JS_BINARY__NODE_WRAPPER" ${JS_BINARY__NODE_OPTIONS[@]+"${JS_BINARY__NODE_OPTIONS[@]}"} -- "$entry_point" ${ARGS[@]+"${ARGS[@]}"} 0<&0
---8<---8<--- End of log ---8<---8<---
INFO: Elapsed time: 1585.004s, Critical Path: 1514.29s
INFO: 8091 processes: 2701 internal, 790 linux-sandbox, 4149 local, 451 worker.
FAILED: Build did NOT complete successfully
```

Is there anyway to enable some kind of logging to narrow it down to what happens?

### Version

Development (host) and target OS/architectures: Linux locally and Linux (nodejs 16.15.0 image in GCP).

Output of `bazel --version`: 5.3.2

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
- rules_js: 1.8.1
- rules_ts: 1.0.2
- rules_swc: 0.19.3 (not utilized atm)

Language(s) and/or frameworks involved:
- TypeScript 4.9.3 running on NodeJS 16.15.0 with [hapi](https://hapi.dev/) as the application server, mocha/chai/sinon for tests.

### Fund our work

- [ ] Sponsor our open source work by donating a [bug bounty](https://opencollective.com/aspect-build/)

**Comment by fredrik-w** [comment]
This is the flags we have in `.bazelrc`:
```text
build --incompatible_strict_action_env
build --nolegacy_external_runfiles
build --noexperimental_check_output_files
build --experimental_remote_merkle_tree_cache --experimental_remote_merkle_tree_cache_size=1000
build --nolegacy_important_outputs
build --experimental_reuse_sandbox_directories
build --incompatible_remote_build_event_upload_respect_no_cache
build --enable_runfiles
test --test_env NODE_ENV=test --test_timeout 600 --test_output=errors --experimental_ui_max_stdouterr_bytes=-1

**Comment by fredrik-w** [comment]
Gah, sorry, thought I was on the rules_ts project, closing this and posting a new one there.
